### PR TITLE
OpenCore: Add CPU Info (MSRs) to SysReport

### PR DESCRIPTION
**Pull request by PMheart**
closes https://github.com/acidanthera/bugtracker/issues/1580

### CODE REVIEW

File: Include/Acidanthera/Library/OcCpuLib.h [hunk 1]
```diff
@@ -160,16 +160,72 @@ typedef struct {
...
+  UINT64                      CpuMsrPlatformInfoValue;
+
+  //
+  // MSR_TURBO_RATIO_LIMIT
+  //
+  BOOLEAN                     CpuHasMsrTurboRatioLimit;
+  UINT64                      CpuMsrTurboRatioLimitValue;
+
+  //
+  // MSR_PKG_POWER_INFO (To be confirmed)
```
Comment (PMheart) [inline Include/Acidanthera/Library/OcCpuLib.h:177]: Add TODO label here.

File: Library/OcDeviceMiscLib/OcDeviceMiscLib.inf [hunk 2]
```diff
@@ -64,6 +65,7 @@
 
 [Sources]
   ActivateHpetSupport.c
+  MsrDump.c
```
Comment (PMheart) [inline Library/OcDeviceMiscLib/OcDeviceMiscLib.inf:68]: Alphabetical order

File: Include/Intel/IndustryStandard/ProcessorInfo.h [hunk 3]
```diff
@@ -56,6 +56,8 @@
 #define MSR_IA32_PERF_CONTROL             0x199
 #define MSR_THERM2_CTL                    0x19D
 
+#define MSR_FSB_FREQ                      0xCD
```
Comment (vit9696) [inline Include/Intel/IndustryStandard/ProcessorInfo.h:59]: MSR_CORE_FSB_FREQ exists in Include/Register/Intel/Msr/CoreMsr.h, use that.

File: Library/OcDeviceMiscLib/MsrDump.c [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+  PrintBuffer (
+    &FileBuffer,
+    &FileBufferSize,
+    "ExternalClock %u\n",
+    CpuInfo->ExternalClock
+    );
+  PrintBuffer (
+    &FileBuffer,
+    &FileBufferSize,
+    "ARTFrequency %Lu, CPUFrequency %Lu, CPUFrequencyFromTSC %Lu, CPUFrequencyFromART %Lu, TscAdjust %Lu, CPUFrequencyFromApple %Lu, CPUFrequencyFromVMT %Lu, FSBFrequency %Lu\n",
```
Comment (vit9696) [inline Library/OcDeviceMiscLib/MsrDump.c:180]: Maybe print line by line.

File: Library/OcDeviceMiscLib/MsrDump.c [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+    &FileBuffer,
+    &FileBufferSize,
+    "MaxId 0x%08X, MaxExtId 0x%08X\n",
+    CpuInfo->MaxId,
+    CpuInfo->MaxExtId
+    );
+  PrintBuffer (
+    &FileBuffer,
+    &FileBufferSize,
+    "PackageCount %u, CoreCount %u, ThreadCount %u\n",
```
Comment (vit9696) [inline Library/OcDeviceMiscLib/MsrDump.c:166]: Maybe print line by line.

**Pull request merged**